podcasts/End of Year/EndOfYear.swift

```diff
@@ -184,3 +184,59 @@ private class FakeViewController: UIViewController {
         onDismiss?()
     }
 }
+
+extension EndOfYear {
+    static let eoyEligibilityDidChange = NSNotification.Name(rawValue: "eoyEligibilityDidChange")
+
+    private class EligibilityChecker {
+        deinit {
+            notifications.forEach { notificationCenter.removeObserver($0) }
+        }
+
+        var isEligible = false
+
+        private let updateQueue = OperationQueue()
+        private let notificationCenter: NotificationCenter
+        private var notifications: [NSObjectProtocol] = []
+
+        init?(notificationCenter: NotificationCenter = .default) {
+            guard FeatureFlag.endOfYear.enabled else { return nil }
+
+            self.notificationCenter = notificationCenter
+
+            startListening()
+            update()
+        }
+
+        private func startListening() {
+            // The notifications to update the state for
+            let notifications: [Notification.Name] = [
+                // Check after a sync succeeds or the user logs into an account
+                ServerNotifications.syncCompleted,
+                .userSignedIn,
+
+                // Check as the user is listening to episodes
+                Constants.Notifications.playbackPaused,
+                Constants.Notifications.playbackEnded,
+                Constants.Notifications.playbackTrackChanged
+            ]
+
+            self.notifications = notifications.map {
+                notificationCenter.addObserver(forName: $0, object: nil, queue: updateQueue) { [weak self] notification in
+                    self?.update()
+                }
+            }
+        }
+
+        func update() {
+            let isEligible = DataManager.sharedManager.isEligibleForEndOfYearStories()
```

Comment:
I think we should only care when the `isEligible` is `false`. If it's `true`, it shouldn't change (even by logging out, given the database data is not erased).

So I think we can add a `guard` statement:

```
            guard !isEligible else {
                return
            }
```